<a href="https://colab.research.google.com/github/adxpillar/DS-Unit-1-Sprint-2-Statistics/blob/master/module1/Adewale_Adeagbo_DS13_LS_DS_121_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [0]:
### YOUR CODE STARTS HERE
import pandas as pd
import numpy as np
from scipy import stats

In [0]:
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

--2020-02-17 22:12:00--  https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18171 (18K) [application/x-httpd-php]
Saving to: ‘house-votes-84.data’

house-votes-84.data 100%[===================>]  17.75K  --.-KB/s    in 0.06s   

2020-02-17 22:12:01 (287 KB/s) - ‘house-votes-84.data’ saved [18171/18171]



In [0]:
df = pd.read_csv('house-votes-84.data',names=['party','handicapped-infants','water-project',
                          'budget','physician-fee-freeze', 'el-salvador-aid',
                          'religious-groups','anti-satellite-ban',
                          'aid-to-contras','mx-missile','immigration',
                          'synfuels', 'education', 'right-to-sue','crime','duty-free',
                          'south-africa'])

In [0]:
df.head()

party handicapped-infants water-project  ... crime duty-free south-africa
0  republican                   n             y  ...     y         n            y
1  republican                   n             y  ...     y         n            ?
2    democrat                   ?             y  ...     y         n            n
3    democrat                   n             y  ...     n         n            y
4    democrat                   y             y  ...     y         y            y

[5 rows x 17 columns]

In [0]:
df = df.replace({"y":1, "n":0, "?": np.NaN})
df.head()

party  handicapped-infants  ...  duty-free  south-africa
0  republican                  0.0  ...        0.0           1.0
1  republican                  0.0  ...        0.0           NaN
2    democrat                  NaN  ...        0.0           0.0
3    democrat                  0.0  ...        0.0           1.0
4    democrat                  1.0  ...        1.0           1.0

[5 rows x 17 columns]

In [0]:
gop = df[df['party'] == 'republican']
dem = df[df['party'] == 'democrat']

In [0]:
df['party'].value_counts()

democrat      267
republican    168
Name: party, dtype: int64

In [0]:
#Probability greater than 0.1
#Hypothesis test for not much difference between republicans and democrats
stats.ttest_ind(gop['water-project'],dem['water-project'],nan_policy='omit')

Ttest_indResult(statistic=0.08896538137868286, pvalue=0.9291556823993485)

𝐻0: There is not much difference between republicans and democrats on water project issue
𝐻1: There is a lot of difference between republicans and democrats on water project issues

Based on the t-stat =0.08896538137868286; and pvalue=0.9291556823993485 we fail to reject the null hypothesis at the 95% confidence level.

In [0]:
gop['water-project'].mean()

0.5067567567567568

In [0]:
dem['water-project'].mean()

0.502092050209205

𝐻0: Dems support buget more than republicans
𝐻1: Dems do not support budget more than republicans 

Based on the t-stat =0.08896538137868286; and pvalue=0.9291556823993485 we fail to reject the null hypothesis that Dems support water project more than republicans at the 95% confidence level.

In [0]:
#dem > GOP
stats.ttest_ind(gop['synfuels'],dem['synfuels'],nan_policy='omit')

Ttest_indResult(statistic=-8.293603989407588, pvalue=1.5759322301054064e-15)

In [0]:
gop['synfuels'].mean()

0.1320754716981132

In [0]:
dem['synfuels'].mean()

0.5058823529411764

𝐻0: Republicans support synthetic fuels more than democrats
𝐻1: Republicans do not support synthetic fuels more than democrats
Based on the t-stat 8.293603989407588, pvalue=0.00000000001.5759322301054064 we reject the null hypothesis and go in favour of the alternative hypothesis at the 95% confidence level.

In [0]:
#gop > dem
stats.ttest_ind(gop['crime'],dem['crime'],nan_policy='omit')

Ttest_indResult(statistic=16.342085656197696, pvalue=9.952342705606092e-47)

In [0]:
gop['crime'].mean()

0.9813664596273292

In [0]:
dem['crime'].mean()

0.35019455252918286

In [0]:
𝐻0: Democrats support crime more than Republicans
𝐻1: Democrats do not support crime control more than republicans
Based on the above t-stat and pvalue we reject the null hypothesis and go in favour of the alternative hypothesis at the 95% confidence level.

In [0]:
gop = df[df['party'] == 'republican']
dem = df[df['party'] == 'democrat']
def hypo_test(gop,dem):
  return stats.ttest_ind(gop,dem,nan_policy='omit')
  